record/blinded_data.go

```diff
@@ -0,0 +1,319 @@
+package record
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+const (
+	// ShortChannelIDType is a record type for the outgoing channel short
+	// ID.
+	ShortChannelIDType tlv.Type = 2
```

Comment:
with the new `tlv.RecordT` and `tlv.OptionalT` types, you can make the tlv number part of the field type itself like: 

```
ShortChannelID tlv.RecordT[tlv.Type2, lnwire.ShortChannelID]
```

See examples [here](https://github.com/lightningnetwork/lnd/pull/8338) and [here](https://github.com/lightningnetwork/lnd/pull/8044)